### PR TITLE
Added note on cache commands over a sudo built container

### LINKED ISSUE #163: make it clear that cache clean should usually be run as root
## Which Document page:

On what page is the problem?
https://www.sylabs.io/guides/3.2/user-guide/build_env.html#cache-folders

### Expected results:

Need to make it more clear (by creating a note or something) that in most circumstance the `cache clean` command will need to be run as root.  See https://github.com/sylabs/singularity/issues/3433

### PR DESCRIPTION
**Pull request by jchavezlavalle**
## Description of the Pull Request (PR):

This one is to fix #163 : When building a container with sudo privileges, the cache commands should consider the use of sudo as well. Default location of cache is `~/.singularity/cache` whereas for sudo privilege is `/root/.singularity/cache`

## This fixes or addresses the following GitHub issues:

- Fixes #163

### CODE REVIEW

**Review by GodloveD — CHANGES_REQUESTED** [review]
I think it could actually be spelled out a little more clearly to the user (actually include the command?)

**Review by GodloveD — APPROVED** [review]
lgtm

File: build_env.rst [hunk 1]
```diff
@@ -39,6 +39,32 @@ Cache commands
...
+    and then cleaning the cache without sudo privilege (``singularity cache clean -a``) will not work, since the default cache location is ``~/.singularity/cache``.
+    In this case you would need to run the clean command with sudo privilege:
+
+    .. code-block:: none
+
+        $ sudo singularity cache clean -a
+        NAME                     DATE CREATED           SIZE             TYPE
+
+        There 0 containers using: 0.00 kB, 0 oci blob file(s) using 0.00 kB of space.
+        Total space used: 0.00 kB
```
Comment (WestleyK) [inline build_env.rst:66]: This looks like the output for `cache list`, not `cache clean`. I would just remove all the output for the `cache clean` command.

File: build_env.rst [hunk 2]
```diff
@@ -39,6 +39,32 @@ Cache commands
 
 Singularity 3.1 comes with new commands for cleaning and listing the cache image files generated.
 
+
+.. note::
+
+    Running the cache commands with sudo privilege will consider cache location as ``/root/.singularity/cache``. The default location for cache without sudo privilege is ``~/.singularity/cache``.
+    Make sure that if you build a container with sudo privilege, you will need to consider the sudo location from the cache, and not the default.
```
Comment (WestleyK) [inline build_env.rst:46]: I would changes this up a bit:

```suggestion
    If you build a container with sudo privilege (like a definition file), you cache will be stored in your root directory (``/root/.singularity/cache``) and not your home directory.

```

File: build_env.rst [hunk 3]
```diff
@@ -39,6 +39,32 @@ Cache commands
...
+
+    For example, running the following command with sudo privilege (considering the sudo privilege location for cache ``/root/.singularity/cache``):
```
Comment (WestleyK) [inline build_env.rst:48]: I would take this out of the `note` block, and change it like this:

```suggestion
For example, running a action (``exec``, ``pull``, ``run``, etc...), command with sudo (or as root), cache will be stored in ``/root/.singularity/cache``, _if the ``SINGULARITY_CACHEDIR`` is **not** set._
```

You can add to that, and changes as you see fit :+1:

File: build_env.rst [hunk 4]
```diff
@@ -39,6 +39,32 @@ Cache commands
...
+
+    .. code-block:: none
+
+        $ sudo singularity cache list
+        NAME                   DATE CREATED           SIZE             TYPE
+        ubuntu_latest.sif      2019-01-31 14:59:32    28.11 Mb         library
+        ubuntu_18.04.sif       2019-01-31 14:58:44    27.98 Mb         library
+
+    and then cleaning the cache without sudo privilege (``singularity cache clean -a``) will not work, since the default cache location is ``~/.singularity/cache``.
```
Comment (WestleyK) [inline build_env.rst:57]: Again, I would remove the spaces before the text, and change it like:

```suggestion
If you set ``SINGULARITY_CACHEDIR`` to something like ``SINGULARITY_CACHEDIR=/tmp/image-cache``, then all cache will be stored in `/tmp/image-cache`. Remember, when you use sudo, to add `-E` flag to your singularity command, like ``sudo -E singularity build test.sif test.def``.
```

And/or add a link to the page where it talks about `SINGULARITY_CACHEDIR`.

*These are just my suggestions, you can change it as you see fit.*

File: build_env.rst [hunk 5]
```diff
@@ -39,6 +39,32 @@ Cache commands
...
+    In this case you would need to run the clean command with sudo privilege:
```
Comment (WestleyK) [inline build_env.rst:58]:
```suggestion
For example cleaning you cache in ``/root/.singularity/cache``:
```

File: build_env.rst [hunk 6]
```diff
@@ -39,6 +39,32 @@ Cache commands
...
+
+    .. code-block:: none
```
Comment (WestleyK) [inline build_env.rst:60]:
```suggestion
.. code-block:: none
```

File: build_env.rst [hunk 7]
```diff
@@ -39,6 +39,32 @@ Cache commands
...
+
+        $ sudo singularity cache clean -a
```
Comment (WestleyK) [inline build_env.rst:62]: And also add a example when `SINGULARITY_CACHEDIR` is set:

```
    # And with 'SINGULARITY_CACHEDIR' set, use the '-E' flag for sudo:
    $ sudo -E singularity cache clean --all
```

File: build_env.rst [hunk 8]
```diff
@@ -39,6 +39,32 @@ Cache commands
...
+    Make sure that if you build a container with sudo privilege, you will need to consider the sudo location from the cache, and not the default.
+
+    For example, running the following command with sudo privilege (considering the sudo privilege location for cache ``/root/.singularity/cache``):
+
+    .. code-block:: none
+
+        $ sudo singularity cache list
+        NAME                   DATE CREATED           SIZE             TYPE
+        ubuntu_latest.sif      2019-01-31 14:59:32    28.11 Mb         library
+        ubuntu_18.04.sif       2019-01-31 14:58:44    27.98 Mb         library
```
Comment (WestleyK) [inline build_env.rst:55]: I would update the output for the cache list command: *Add the summary after the cache*

```
    $ sudo singularity cache list 
NAME                     DATE CREATED           SIZE             TYPE
ubuntu_latest.sif        2019-05-07 12:53:14    28.11 MB         library
ubuntu_18.04.sif         2019-05-07 12:53:24    27.98 MB         library

There 2 containers using: 56.09 MB, 0 oci blob file(s) using 0.00 kB of space.
Total space used: 56.09 MB
```

**Review by WestleyK — COMMENTED** [review]
Sorry I'm kinda late for the review, maybe make another PR fixing the comments?... These are just some suggestions :+1:

**Pull request merged**